.github/workflows/sanitizer-build-and-test.yaml

```
@@ -0,0 +1,157 @@
+name: Sanitizer test
+on:
+  schedule:
+    # run daily 20:00 on master branch
+    - cron: '0 20 * * *'
+  push:
+    branches:
+      - prerelease_test
+jobs:
+  sanitizer:
+    name: PG${{ matrix.pg }} ${{ matrix.name }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: ["Sanitizer"]
+        pg: ["12.7", "13.3"]
+        os: ["ubuntu-20.04"]
+        build_type: ["Debug"]
```

I do not think you need the single-option alternatives in the workflow.
```suggestion
      matrix:
        pg: ["12.7", "13.3"]
```

---

How will those lines work if it's not in the workflow?
```
jobs:
  sanitizer:
    name: PG${{ matrix.pg }} ${{ matrix.name }} ${{ matrix.os }}
```

---

You have to either define variables for them in the `env` section or write them there literally. The matrix is intended to test several different combinations of options and basically computes the Cartesian product of all the entries.

---

I would put those in `env` but I think `env` doesn't work until you actually execute some real code, e.g. in the `name:` section environment variables wouldn't work. I wanted some workflow native variables so that we can deduplicate strings such as "ubuntu 20.04". I don't want to manually hunt for all "ubuntu 20.04" strings later when we move to a different version for some reason. Is there a way to define such workflow native variables outside of the matrix?

---

No idea. I would test with the environment variable and if it does not work, type them literally (and as a consequence, hunt them down when they need to change).